hcipy/aperture/generic.py

```diff
@@ -119,6 +121,37 @@ def func(grid):
 
 	return func
 
+def irregular_polygon_aperture(vertices):
+	'''Make an irregular polygonal aperture.
+
+	Parameters
+	----------
+	vertices : iterable of length 2 iterables
+		The vertices of the polygon.
+
+	Returns
+	-------
+	Field generator
+		The generator for the polygonal aperture.
+	'''
+	p = Path(vertices)
+
```

Comment:
Should we add a check here to see if it creates a valid polygon?

1. Is the path closed?
2. What happens when the vertices cross each other?

Or is this over kill at the moment? We could always add that in a future PR.

Reply:
1. The path is automatically closed by matplotlib.
2. For four vertices on a square, this is what happens if you have them in the right order:
![image](https://user-images.githubusercontent.com/38925765/183223564-ae337518-3591-4267-849d-0d83b920eb9d.png)
and this for the wrong order:
![image](https://user-images.githubusercontent.com/38925765/183223736-e5a311ee-108f-4514-83ed-2c9bfa00256a.png)

Reply:
Okay. Then this is resolved.